.devcontainer/Dockerfile

```diff
@@ -1,27 +1,43 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
-FROM mcr.microsoft.com/devcontainers/miniconda:3 AS base
+FROM mcr.microsoft.com/vscode/devcontainers/base AS base
```

Comment:
This is the main change.
It reduces the initial layer to something generally pretty small.
However, it doesn't include miniconda, so the next step is to reinstall miniconda.
However, we do that along with preparing the mlos conda env all in one step since conda uses hardlinks to save space and combining those into a single step allows tar (the tool used by docker) retain that optimization rather than making copies across docker layers.